From 29e0f0b28a52db83b7d4e7fb7e8ebcd4995f4c3b Mon Sep 17 00:00:00 2001
From: Rui Matos <tiagomatos@gmail.com>
Date: Mon, 11 Feb 2013 18:07:06 +0000
Subject: region: Show/hide IBus sources config button

Instead of just making it sensitive/unsensitive. Quoting from the bug
report:

The problem is that it isn't clear to the user why the settings button is
insensitive for keyboard layouts - they'll be asking "why can't I ever use the
settings?"

https://bugzilla.gnome.org/show_bug.cgi?id=692006

Index: gnome-control-center-3.6.3/panels/region/gnome-region-panel-input.c
===================================================================
--- gnome-control-center-3.6.3.orig/panels/region/gnome-region-panel-input.c	2013-09-11 10:27:26.979533097 -0400
+++ gnome-control-center-3.6.3/panels/region/gnome-region-panel-input.c	2013-09-11 10:33:09.507518772 -0400
@@ -1104,7 +1104,7 @@
   gtk_widget_set_sensitive (show_button, index >= 0);
   gtk_widget_set_sensitive (up_button, index > 0);
   gtk_widget_set_sensitive (down_button, index >= 0 && index < n_active - 1);
-  gtk_widget_set_sensitive (settings_button, settings_sensitive);
+  gtk_widget_set_visible (settings_button, settings_sensitive);
 }
 
 static void
